tensorflow/compiler/xla/service/pattern_matcher.h

```diff
@@ -2157,6 +2225,22 @@ XLA_NULLOP_PATTERN(PartitionId)
 XLA_NULLOP_PATTERN(ReplicaId)
 #undef XLA_NULLOP_PATTERN
 
+// Matches any of a set of unary ops including the operand and optionally
+// returns the index of the matched op.
+template <typename Arg>
+inline auto UnaryOpAnyOfOrOperand(Arg&& arg, absl::Span<const HloOpcode> ops,
+                                  int* match_index = nullptr) {
```

Comment:
I think you shouldn't need match_index.  You can just write

```
HloInstruction* instr;
Match(foo, UnaryOpAnyOfOrOperand(Op(&instr), {kAdd, kSubtract})))
```

and then you can inspect `instr` to see its opcode?